### PR TITLE
Airlock UI: create review VM

### LINKED ISSUE #2735: Create Review VM from the Airlock UI screen
## Description

As a TRE Airlock Manager
I want to have a button next to the Airlock Request to spin up the Review VM
So that I don't have to remember which workspace and workspace service to create it in and I don't have to copy SAS URL

### Configuration

We propose to add an additional nested block to Research Workspace (=base workspace) properties for configuring the Airlock Review VMs. 

This could look like this: 

```
{
    "id": "<id>",
    "templateName": "tre-workspace-base",
    "templateVersion": "0.3.28",
    "properties": {
        "enable_airlock": true,
        "airlock_review_config": {
            "import": {
                "workspace_id": "<guid>",  // ID for the already deployed Import Review Workspace to use for reviews
                "workspace_service_id": "<guid>",  // ID for the already deployed Workspace Service to deploy review VMs to
                "user_resource_template_name": "guacamole-azure-import-reviewvm"  // the user resource template to create
            },
            "export": {
                "workspace_serivce_id": "<guid>",  // ID for the already deployed Workspace Service to deploy review VMs to
                "user_resource_template_name": "guacamole-azure-import-reviewvm"  // the user resource template to create
            },
        },
    // other properties as normal
    },
```

The `export` section does not have `workspace_id` because for export it will always be the Research Workspace the data is exported from. 
The corresponding resources need to already be deployed before this configuration can be set, which is why we should recommend users to issue an update on an existing resource if they wish to configure Airlock Review. 

### UI screens

We then propose to add a button / screen on the UI next to the Airlock Review requests ([updated in Figma](https://www.figma.com/file/Ncl8TJsFf2WKf3frrqTbtq/AzureTRE-OSS-UI?node-id=306%3A21280)): 

image

image

image

The button will be greyed out if the above configuration is not set for the workspace, with a message for the user to ask their TRE admin to configure this for them. 

image

If there already is a review VM created for the request by this user, it could show a link to that created review VM. 
If the review VM was created by another user, it could show a message saying that another user is currently reviewing this request. For this, the `review_user_resources` field in Airlock Request object is used (see below).

image

### API endpoint

Add a corresponding API endpoint that will create a review User Resource

`/api/workspaces/{workspace_id}/requests/{airlock_request_id}/create-review-user-resource`

This will accept the Research Workspace ID and Request ID as parameters, from the properties outlined above will determine in which Workspace and Workspace Service to create them in, will create the user resource and return the Operation to watch, which will be tracked in the usual way on the UI. 

The API endpoint will also update the Request object with the information on this VM: 

```
{
    "id": "45e1d9d8-f669-47bf-b0eb-a11bb34aa456",
    "workspaceId": "26f41c2a-2bed-4f78-a745-c41ca8f326fe",
    "requestType": "export",
    "files": [
        {
            "name": "research_result.txt",
            "size": 7
        }
    ],
    "review_user_resources": [
        {
            "id": "<guid>",
            "workspace_id": "<guid>",
            "workspace_service_id": "<guid>"
        }
    ],
    // other properties as normal
}
```

## Acceptance criteria

- [x] #2736
- [x] #2737
- [x] #2738
- [ ] #2830

### LINKED ISSUE #2738: UI for Airlock Review VM
https://www.figma.com/file/Ncl8TJsFf2WKf3frrqTbtq/AzureTRE-OSS-UI?node-id=1293%3A27999

### PR DESCRIPTION
**Pull request by jjgriff93**
# Resolves
#2738

## What is being addressed

Adds ability to deploy a review Vm in airlock review UI and open to view airlock request data securely in the review workspace

## How is this addressed

- Redesigned review experience as multi-step modal
- Implemented create-review-vm endpoint
- Display relevant resource state (created, not created, creating) 
- Added workspace scope option to componentManager hook so we can check for updates on resource that's in a different workspace to the current context (needed for import review Vms that reside in a different workspace)
- Changed create-review-vm api endpoint to return airlock request as well as create vm operation to reduce calls and latency on front-end and potential state mismatch

image

image

image

Note: this only supports single review VMs currently. A follow-up story adds support for multiple reviewers' Vm resources: #2764

### CODE REVIEW

File: api_app/_version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "0.4.54"
+__version__ = "0.4.57"
```
Comment (tamirkamara) [inline api_app/_version.py:1]: Commenting here as there's no better place for this:
1. Can you update the changelog
2. [nit] I think it's best to use a vector image (and not a raster one). It should be simple enough to find an SVG version of it...

Comment (jjgriff93) [reply api_app/_version.py:1]: Yep good shout thanks, have addressed both

**Review by damoodamoo — APPROVED** [review]
LGTM!

**Pull request closed without merging**